test.sh

```diff
@@ -3,40 +3,89 @@ if [ -z "$MATRIX_SCALA" ]; then
-pushd play-scala-chatroom-example       && scripts/test-sbt && popd
-pushd play-scala-compile-di-example     && scripts/test-sbt && popd
-pushd play-scala-fileupload-example     && scripts/test-sbt && popd
-pushd play-scala-forms-example          && scripts/test-sbt && popd
-pushd play-scala-grpc-example           && scripts/test-sbt && popd
-pushd play-scala-hello-world-tutorial   && scripts/test-sbt && popd
-pushd play-scala-isolated-slick-example && scripts/test-sbt && popd
-pushd play-scala-log4j2-example         && scripts/test-sbt && popd
-pushd play-scala-macwire-di-example     && scripts/test-sbt && popd
-pushd play-scala-rest-api-example       && scripts/test-sbt && popd
-# Uses libsodium
-pushd play-scala-secure-session-example && scripts/test-sbt && popd
-pushd play-scala-slick-example          && scripts/test-sbt && popd
-pushd play-scala-starter-example        && scripts/test-sbt && popd
-pushd play-scala-streaming-example      && scripts/test-sbt && popd
-# uses vanilla sbt
-pushd play-scala-tls-example            && scripts/test-sbt && popd
-pushd play-scala-websocket-example      && scripts/test-sbt && popd
+# Initialize variables
+sample=""
+backend_server="pekko"
+build_tool="sbt"
+
+# Parse command line arguments
+while [[ "$#" -gt 0 ]]; do
+  case $1 in
+    --build-tool=*) build_tool="${1#*=}" ;;  # Extract the value right of the equal sign
+    --backend-server=*) backend_server="${1#*=}" ;;  # Extract the value right of the equal sign
+    --sample=*) sample="${1#*=}" ;;  # Extract the value right of the equal sign
+    *) echo "Unknown parameter passed: $1"; exit 1 ;;
+  esac
+  shift  # Move to next argument
+done
+
+function buildSample() {
+    local dir=$1
+    pushd $dir
+    if [[ $build_tool == "gradle" ]]; then
+      if [ -f scripts/test-gradle ]; then
+        scripts/test-gradle
+      else
+        echo "+-----------------------------------------+"
+        echo "| Executing tests using Gradle (Default)  |"
+        echo "+-----------------------------------------+"
+        ./gradlew -Dscala.version="$MATRIX_SCALA" check
+      fi
+    elif [[ $build_tool == "sbt" ]]; then
+      if [[ $backend_server == "netty" ]]; then
+          find . -name build.sbt | xargs perl -i -pe 's/\/\/\.enablePlugins\(PlayNettyServer\)/.enablePlugins(PlayNettyServer)/g'
+      fi
+      if [ -f scripts/test-sbt ]; then
+        scripts/test-sbt
+      else
+        echo "+----------------------------+"
+        echo "| Executing tests using sbt  |"
+        echo "+----------------------------+"
+        sbt ++$MATRIX_SCALA test
+      fi
+    fi
+    popd
+}
+
+if [ -n "$sample" ]; then
+  # Sample is set and not empty
+  buildSample $sample
+else
+  # sample not set, therefore run all samples
+  buildSample play-java-chatroom-example
+  buildSample play-java-compile-di-example
+  buildSample play-java-chatroom-example
+  buildSample play-java-compile-di-example
```

Comment:
These 4 lines are duplicated, 2 can be removed (probably a copy/paste "error" :wink:)

Reply:
Ooops, sorry